.php-cs-fixer.dist.php

```diff
@@ -0,0 +1,34 @@
+<?php
+/**
+ * PHP-CS-Fixer configuration
+ *
+ * (c) Alley <info@alley.co>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @package wp-find-one
```

Comment:
I'd recommend including an SPDX expression containing this basic metadata as a quick human-and-machine-readable reference.

https://github.com/david-a-wheeler/spdx-tutorial#spdx-license-expressions-in-source-code-files
https://spdx.github.io/spdx-spec/

```suggestion
 * Find One: PHP-CS-Fixer configuration
 *
 * Copyright (c) 2022, Alley and the Find One contributors
 * SPDX-License-Identifier: GPL-2.0-or-later
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @package wp-find-one
```
```suggestion
 * PHP-CS-Fixer configuration
 *
 * (c) Alley <info@alley.co>
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @package wp-find-one
```

Reply:
This construction of comment is modeled off of file headers of other PHP libraries, with a few tweaks to match the WordPress core standards for file headers. For example:

- https://github.com/Seldaek/monolog/blob/60ad5183b5e5d6c9d4047e9f3072d36071dcc161/src/Monolog/Logger.php#L1-L10
- https://github.com/sebastianbergmann/phpunit/blob/77354693683de022702c9b95e7fe8d840476f82c/src/Framework/TestSuite.php#L1-L10

So I'm inclined to leave off an SPDX header for consistency with those.

As a personal aside, while the SPDX header might be machine-readable, it doesn't come off to me as super human-readable. If I don't know what SPDX is (I had to Google it), the line won't mean anything to me. Moreover, it seems redundant: Why include both the license name and the invitation to view `LICENSE` in the same block?

Reply:
> This construction of comment is modeled off of file headers of other PHP libraries, with a few tweaks to match the WordPress core standards for file headers.

> So I'm inclined to leave off an SPDX header for consistency with those.

Consistency bolsters familiarity, but without an eye towards the distinctly separate principle of actual documented best practices, consistency reinforces a copy-pasta status quo.

I know we can do better. To be quite honest, I think those projects do a terrible job of explaining their users' rights in those headers. They might as well not include a header at all if they're going to point to a file somewhere else. Which, by the way, I think is totally fine! It's the "I won't tell you the license, go find it" that I find frustrating. How do you know they didn't copy that "look somewhere else" wording without asking whether it was a good idea? 

Perhaps that works out for those respected monolithic projects whose components are unlikely to be distributed in fragments, but when the library primarily consists of a single file which can easily be lifted into another project, I would argue that we should _expect_ that the license will never be recirculated in those cases, so the message about some other file becomes useless upon later reference.

> While the SPDX header might be machine-readable, it doesn't come off to me as super human-readable.

While `GPL-2.0-or-later` is a strict string, its meaning is obvious to many. If someone is unfamiliar, the benefit of a strict string is that it's easily referenced in a search engine query e.g. `license GPL-2.0-or-later`.

> If I don't know what SPDX is (I had to Google it), the line won't mean anything to me.

That seems to me like an exaggeration.

Sometimes we know a thing before we know its name. To me, the addition of the name `SPDX` provided a name for a set of rules I never thought about before but which all of us, as developers, use all the time.

Reply:
> without an eye towards the distinctly separate principle of actual documented best practices, consistency reinforces a copy-pasta status quo

Where is it documented that SPDX is a best practice, other than by the SPDX workgroup itself? Got links? 😄

> when the library primarily consists of a single file which can easily be lifted into another project, I would argue that we should expect that the license will never be recirculated in those cases, so the message about some other file becomes useless upon later reference

This convinces me.

> If I don't know what SPDX is (I had to Google it), the line won't mean anything to me.

I'm not exaggerating. Yes, I can make some guesses based on familiar strings, but I wouldn't want to bet the compliance of my code with a license on it. "I won't tell you the license, go find it" is curt but pretty unambiguous in telling you "the answer is not here."

In any case, I'm not quite prepared to make the call by myself to switch to SPDX, so I'll bring this discussion to the open source working group to find an improvement, SPDX or otherwise.

Reply:
Press release: [SPDX Becomes Internationally Recognized Standard for Software Bill of Materials - Linux Foundation](https://linuxfoundation.org/press-release/spdx-becomes-internationally-recognized-standard-for-software-bill-of-materials/)

> _Backed by many of the world’s largest companies for more than a decade, SPDX formally becomes an internationally recognized ISO/IEC JTC 1_ _standard during a transformational time for software and supply chain security_

Also see: [Solving License Compliance at the Source: Adding SPDX License IDs - Linux Foundation](https://www.linuxfoundation.org/blog/solving-license-compliance-at-the-source-adding-spdx-license-ids/)

SPDX identifiers are included in the Wikipedia sidebars for software licenses:

- https://en.wikipedia.org/wiki/GNU_General_Public_License
- https://en.wikipedia.org/wiki/MIT_License
- https://en.wikipedia.org/wiki/Unlicense

Specifically in the context of WordPress:

- https://github.com/WordPress/gutenberg/pull/4338
- https://core.trac.wordpress.org/ticket/43032

In terms of source file headers specifically… I think that's certainly more subjective, since file headers can lend a bit of character to a project. But at the very least I would expect a file header, if there is one, would include the name of the license. Rather than winging it, we have the official license naming schemes provided by SPDX to lean on. The "Solving License Compliance at the Source" article linked above highlights some early examples.

Reply:
Thanks for the links!

The content of source file header, I thought, was the heart of the discussion 😄 But using an SPDX identifier for the license _if_ the header includes the name of the license certainly makes sense to me.